evmap-derive/tests/failing/lib.rs

```diff
@@ -0,0 +1,25 @@
+use evmap_derive::ShallowCopy;
+use evmap::shallow_copy::ShallowCopy as _;
+
+#[derive(ShallowCopy)]
+struct Broken {
+	f1: i32,
+	f2: std::cell::Cell<()>,
+}
+
+#[derive(ShallowCopy)]
+struct AlsoBroken(i32, std::cell::Cell<()>);
+
+#[derive(ShallowCopy)]
+struct Generic<T> {
```

Comment:
Hmm, why should this not work? Seems reasonable to me as long as the `T: ShallowCopy` bound is added?

Reply:
In fact, don't we have the _exact_ same test passing in the other test file?

Reply:
We do, and it does! It's just there as a setup for the failing usage in the function below. The derive automatically adds that bound, which is pretty much just what I'm testing here.

Reply:
Ah, you're right, I missed that the error is on _use_, not on declaration. Ignore me — clearly today is not my day :sweat_smile: